drivers/can/can_mcan.c

```diff
@@ -841,6 +841,10 @@ int can_mcan_send(const struct device *dev,
 		return -ENETUNREACH;
 	}
 
+	if (frame->fd && (can->cccr & (CAN_MCAN_CCCR_FDOE | CAN_MCAN_CCCR_BRSE)) == 0) {
```

Comment:
I just copied the same bits that were set by the mode set function. I probably should actually split it, so frame->brs is checked against that (Since its possible to have fd without baud rate changing enabled at the HW level.

Reply:
I split the checks, and added logs too.